packages/x-utils/src/object.ts

```diff
@@ -94,6 +94,32 @@ export function cleanUndefined<T>(obj: T): T {
       );
 }
 
+/**
+ * Creates an object picking only the properties whose values are not:
+ * - `undefined`.
+ * - `null`.
+ * - an empty string.
+ * - an empty array.
+ *
+ * @param obj - The object from whom pick the values.
+ * @returns A new object with the not empty properties of the source object.
+ * @public
+ */
+export function cleanEmpty<T extends Record<string, unknown>>(obj: T): T {
+  return reduce(
+    obj,
+    (pickedObject, key, value) => {
+      if (isObject(value)) {
```

Comment:
Should we clean also empty objects? It is not specified in the task, but i think it is the same case as the empty array. Doing that would also complicate the logic a little bit (one more `if`), as you mind end up removing all the properties from a child object, transforming it into an empty object that should be removed then.

I guess that is not needed for the http-client use case, as it flattens the parameters for the GET requests

Reply:
Sure, I totally agree! I've added the check so the empty objects are also cleaned.

Reply:
Good point! I added it and also a comment to fix the function to clean nested empty arrays too in the future.